### DISCUSSION TITLE
Case study proposal - Ecommerce web application using eShoponWeb

### DISCUSSION THREAD

**Discussion by srini1978** [Green-Software-Foundation/sci#225] [SCI feedback]
As we are starting to identify case studies, one option we can use is to take a popular ecommerce open source code and calculate SCI for it. eShop is a popular one that is Asp.net core based. There are multiple variants - using containers, using Managed web services

https://github.com/dotnet-architecture/eShopOnWeb

SCI = (E*I) +M per R

Proposal
1. Deploy the application in Web based services on any cloud provider . For e.g App service in Azure  with backend SQL server. For managed azure services, we will get the Vcore , RAM specs (# of cores , 3 GB/4 GB RAM). Calculate M based on spec sheet . Need to find out if we can get the thermal design power TDP value from the specs.
2. Run the application in Azure and  calculate CPU consumption, Memory consumption and Disk consumption. Use the combination of these values to come up with E value. 
3.  Use Watt time API to get the marginal carbon intensity value of I for the data center in which the application is deployed.
4. R will be the functional unit here. In this case, we can consider R as "per number of users". For example SCI value of x per 100 users or 1000 users depending on the non functional requirements of the application.

@Henry-WattTime  @atg-abhishek

**Comment by atg-abhishek** [comment]
I like this idea @srini1978 will you be leading the charge on trying out this setup and making the calculations?

**Reply by srini1978** [reply]
yes @atg-abhishek  I will take this forward

**Comment by atg-abhishek** [comment]
@Sealjay suggests maybe using something like [Digital Ocean](https://www.digitalocean.com/) as well

**Comment by srini1978** [comment]
One of the key inputs that we need to calculate E is the server name or server family on which the software is running.  This information is not publicly available for most of the cloud providers. What are the datasets that we have already that can help get this information

**Reply by mrchrisadams** [reply]
These are available in machine readable form in this repo already - there are many assumptions in the dataset, but you can at least see the basis of the assumptions, and how to provide more accurate and precise information for calculations.

https://github.com/cloud-carbon-footprint/cloud-carbon-coefficients/tree/main/output

**Reply by srini1978** [reply]
> These are available in machine readable form in this repo already - there are many assumptions in the dataset, but you can at least see the basis of the assumptions, and how to provide more accurate and precise information for calculations.
> 
> https://github.com/cloud-carbon-footprint/cloud-carbon-coefficients/tree/main/output

@mrchrisadams  thanks for this. Are the units for the embodied emissions output in grams co2eq?

**Reply by davidmytton** [reply]
kgCO2e

**Reply by srini1978** [reply]
@davidmytton  @mrchrisadams  Thanks for this dataset. While I can infer that I can use the dataset around embodied emissions directly, I am not exactly clear how to use the other dataset which pertains to "use". there is a min value and max value per processor family . how we do we use that ?

**Reply by davidmytton** [reply]
The embodied emissions can be used directly because they're fixed - incurred once at assessed at the time/place of manufacture. The use stage is represented in terms of energy because the emissions will vary depending on the location (the cloud region, in the case of the Cloud Carbon Footprint project), which is why the min/max watts are provided. You'd need to use your own emissions factors to convert the energy values into kgCO2e.

**Reply by srini1978** [reply]
@davidmytton  another question on the "use stage" dataset. There is a column that says GB/chip. What does this column value indicate and is this in KwH as well ?

**Reply by davidmytton** [reply]
Yeah it's not entirely clear how this is used from the source file, but it's part of how the CCF does the emissions calculation based on how much energy is used for memory. The GB/chip number is how much memory there is per CPU socket for each CPU architecture. The average is then calculated for all instances for the cloud provider to give a fixed kWh/GB. That can then be allocated based on assigned memory for each instance type.

Using Azure as an example, the amount of memory is given an energy factor at https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/blob/04f2be1645f9d1f6a5569ea6469e275a3ec143f8/packages/azure/src/domain/AzureFootprintEstimationConstants.ts#L99 which is then multiplied by the region specific carbon factor based on usage.

**Reply by davidmytton** [reply]
> What does this column value indicate and is this in KwH as well ?

The value in https://github.com/cloud-carbon-footprint/cloud-carbon-coefficients/blob/main/coefficients.ipynb is in GB. It's not converted to kWh until it's used in https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/blob/04f2be1645f9d1f6a5569ea6469e275a3ec143f8/packages/azure/src/domain/AzureFootprintEstimationConstants.ts#L99

**Comment by ElmarBorgmeier** [comment]
It's great that you are actually putting the SCI into practice! 

From your approach, it becomes evident that the SCI will rely on the server type and data center as much as on the software. So the SCI will effectively be a rating of the combination of software+server+data center.

Well, let's see what you can achieve and what the results tell us. Maybe, if you publish all values including I and M, it will be possible to separate the influence of the software from hardware/data center.

I'll stay tuned!

**Comment by Henry-WattTime** [comment]
List of global annual marginal emissions factors: https://unfccc.int/climate-action/sectoral-engagement/ifis-harmonization-of-standards-for-ghg-accounting/ifi-twg-list-of-methodologies

**Comment by srini1978** [comment]
@Henry-WattTime Thanks for sharing the link that has the data for marginal intensity. I believe this is the link right ? - https://unfccc.int/sites/default/files/resource/IFI%20Default%20Grid%20Factors%202021%20v3.1_unfccc.xlsx.  Quick question , If i look at the dataset then I can see that for India it is 951 gCo2/kwH. Am I interpreting it right ?

**Reply by Henry-WattTime** [reply]
Yes! That is correct.

**Comment by srini1978** [comment]
@Henry-WattTime  @atg-abhishek  Have uploaded the results and case study document in the issues tab. #227

**Reply by atg-abhishek** [reply]
Thank you @srini1978!
